src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1510,7 +1510,8 @@ bool AbstractWriteLog<I>::check_allocation(
     /* We need one free log entry per extent (each is a separate entry), and
      * one free "lane" for remote replication. */
     if ((m_free_lanes >= num_lanes) &&
-        (m_free_log_entries >= num_log_entries)) {
+        (m_free_log_entries >= num_log_entries) &&
+        (m_bytes_allocated + bytes_allocated < m_bytes_allocated_cap)) {
```

Comment:
Why `<` now that `m_bytes_allocated_cap` is reduced?  I think it should be `<=`.

Reply:
Let's express it as `m_bytes_allocated_cap >=  m_bytes_allocated + bytes_allocated` for consistency with `m_free_lanes` and `m_free_log_entries` checks.